aten/src/ATen/native/cuda/UpSampleNearest2d.cu

```diff
@@ -233,7 +237,9 @@ static void upsample_nearest2d_backward_out_cuda_template(
   unsigned int n = grad_input.numel() / nbatch;
   dim3 bdim{std::min<unsigned int>(
       at::cuda::getCurrentDeviceProperties()->maxThreadsPerBlock, MAX_THREADS)};
-  dim3 gdim{cuda::ATenCeilDiv(n, bdim.x)};
+  dim3 gdim{std::min<unsigned int>(
+      at::cuda::getCurrentDeviceProperties()->maxGridSize[0],
+      cuda::ATenCeilDiv(n, bdim.x))};
```

Comment:
same here, you are not handling it correctly in the kernel, so don't truncate grid dim here just to silently produce wrong results